File: lab1/lab1_1.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <sstream>
#include <string>

using namespace std;

void SelectionSort(vector<int>& A) {
    int n = (int)A.size();

    for (int i = 0; i < n - 1; i++) {
        int maxIndex = i;

        for (int j = i + 1; j < n; j++) {
            if (A[j] > A[maxIndex]) {
                maxIndex = j;
            }
        }

        swap(A[i], A[maxIndex]);
    }
}

int main() {
    vector<int> A;
    string line;

    getline(cin, line);
    stringstream ss(line);

    int x;
    while (ss >> x) {
        A.push_back(x);
    }

    SelectionSort(A);

    for (int i = 0; i < (int)A.size(); i++) {
        cout << A[i] << " ";
    }

    return 0;
}

File: lab1/lab1_1.py
def SelectionSort(A):
    n = len(A)

    for i in range(n - 1):
        max_index = i

        for j in range(i + 1, n):
            if A[j] > A[max_index]:
                max_index = j

        A[i], A[max_index] = A[max_index], A[i]


A = list(map(int, input().split()))

SelectionSort(A)

print(*A)
